### DISCUSSION TITLE
Whats the best approach to configure cells in TCA UIKit

### DISCUSSION THREAD

**Discussion by abdelmajidrajad** [pointfreeco/swift-composable-architecture#433] [Q&A]
I have a big UI issue (collection not reloading) while using viewStore on each cell. 
I am sending event after dequeuing the cell, is there any better approach ? 

```swift 
public func configure(_ store: Store<CellState, CellAction>) {
        viewStore = ViewStore(store)
        viewStore.send(.load) // to load like images and video images.
        viewStore.producer.state.startWithValues { values in 
         // subscription 
        }
}
```

**Comment by RabugenTom** [comment]
Hey @abdelmajidrajad, remember that `ViewStore` is an `ObservableObject` so make sure to use `@ObservedObject var viewStore: ViewStore` in your cells if you're not using `WithViewStore`.

**Reply by abdelmajidrajad** [reply]
I'm not playing with SwiftUI, but UIKit

**Reply by lukeredpath** [reply]
You'll still need to observe the view stores changes if you aren't already, it's just a bit more explicit without the SwiftUI magic. Are you subscribing to the view store's publisher when your view controller view has loaded?

**Reply by maximkrouk** [reply]
You may just treat your Images as UIComponents (but urls are state) and load them in the UILayer. I recommend Nuke framework for dealing with images. Also UICollectionViewPrefetchingDataSource is a nice place to prefetch all the data you need, Nuke also manages image preheating and progressive loading (image preheating may be extracted to reducer's environment btw)

**Reply by abdelmajidrajad** [reply]
@lukeredpath yes, I subscribe to the view store on configure function. 
@maximkrouk  thanks, does Nuke provide a shared manager to load images from URL ?

**Reply by abdelmajidrajad** [reply]
I end up using third party framework sdwebimage  after subscribing to the viewStore directly on cell code not in the environment. 
Still want to get things better in UIKit but conclude by that. holding image data in state & load images make things worse.

**Reply by maximkrouk** [reply]
@abdelmajidrajad yep, Nuke provides multiple loading instruments for your images, performant cache, imagePreheater for preloading, loader for UIImageViews and beautiful API. I like it much better, than SDWebImage API. I still use SDWebImageWebPDecoder for decoding webp images (Nuke plugin is available via CocoaPods, but I use SwiftPM, so currently I have a mix of Nuke and SD), but gonna migrate to Nuke-only asap

**Reply by abdelmajidrajad** [reply]
Thanks @maximkrouk, I will migrate to it as soon as possible.

**Reply by fabstu** [reply]
@maximkrouk Have you also tried Kingfisher? First I've heard of Nuke, but I'm pretty new to this so it doesn't mean much I guess.
